netlogo-gui/src/main/window/AutoSuggest.scala

```diff
@@ -0,0 +1,218 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.window
+
+import org.nlogo.core.DefaultTokenMapper
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.collection.immutable.TreeSet
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+/**
+  * Builds the trie from commands and reporters and provides fuctions to get suggestions.
+  * Special care should be taken as everything is inserted and retrieved in lower case.
+  */
+class AutoSuggest {
+  val commandNames = DefaultTokenMapper.allCommandNames
+  val reporterNames = DefaultTokenMapper.allReporterNames
+
+  val trie = new TrieNode()
+  for(commandName <- commandNames) {
+    trie.append(commandName.toLowerCase)
+  }
+  for(reporterName <- reporterNames) {
+    trie.append(reporterName.toLowerCase)
+  }
+
+  def editDistance(s1: String, s2: String): Int = {
+    val memo = scala.collection.mutable.Map[(List[Char],List[Char]),Int]()
+    def min(a:Int, b:Int, c:Int) = Math.min( Math.min( a, b ), c)
+    def sd(s1: List[Char], s2: List[Char]): Int = {
+      if (memo.contains((s1,s2)) == false)
+        memo((s1,s2)) = (s1, s2) match {
+          case (_, Nil) => s1.length
+          case (Nil, _) => s2.length
+          case (c1::t1, c2::t2)  => min( sd(t1,s2) + 1, sd(s1,t2) + 1,
```

Comment:
I prefer not to have spaces between the open parens of a method invocation and its first argument. If you want to space out the arguments it's totally acceptable to move each argument to its own line.